### PR TITLE
chore: migrate to new deno_core and metrics

### PR DESCRIPTION
**Pull request by mmastrac**
- Uses the new OpMetrics system for sync and async calls
- Partial revert of #21048 as we moved Array.fromAsync upstream to deno_core

### CODE REVIEW

File: cli/tests/unit/metrics_test.ts [hunk 1]
```diff
@@ -1,93 +0,0 @@
-// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
-import { assert, assertEquals } from "./test_util.ts";
-
-Deno.test(async function metrics() {
```
Comment (mmastrac) [inline cli/tests/unit/metrics_test.ts:4]: This is tested in deno_core now.

File: cli/tests/unit/timers_test.ts [hunk 2]
```diff
@@ -412,19 +412,6 @@ Deno.test(function clearTimeoutAndClearIntervalNotBeEquals() {
   assertNotEquals(clearTimeout, clearInterval);
 });
 
-Deno.test(async function timerMaxCpuBug() {
```
Comment (mmastrac) [inline cli/tests/unit/timers_test.ts:415]: I couldn't think of a good way to replicate this test without using metrics... thoughts?

File: cli/tests/integration/run_tests.rs [hunk 3]
```diff
@@ -3425,11 +3425,6 @@ itest!(unstable_ffi_19 {
   exit_code: 70,
 });
 
-itest!(future_check2 {
```
Comment (mmastrac) [inline cli/tests/integration/run_tests.rs:3428]: Was effectively doing and testing nothing.

File: cli/args/flags.rs [hunk 4]
```diff
@@ -398,6 +398,7 @@ pub struct Flags {
   pub config_flag: ConfigFlag,
   pub node_modules_dir: Option<bool>,
   pub vendor: Option<bool>,
+  pub enable_op_summary_metrics: bool,
```
Comment (mmastrac) [inline cli/args/flags.rs:401]: This is a flag but isn't currently exposed on the CLI args yet.

File: cli/tsc/dts/lib.deno.ns.d.ts [hunk 5]
```diff
@@ -3744,29 +3744,7 @@ declare namespace Deno {
...
-   * > console.table(Deno.metrics())
-   * ┌─────────────────────────┬────────┐
-   * │         (index)         │ Values │
-   * ├─────────────────────────┼────────┤
-   * │      opsDispatched      │   3    │
-   * │    opsDispatchedSync    │   2    │
-   * │   opsDispatchedAsync    │   1    │
-   * │ opsDispatchedAsyncUnref │   0    │
-   * │      opsCompleted       │   3    │
-   * │    opsCompletedSync     │   2    │
```
Comment (bartlomieju) [inline cli/tsc/dts/lib.deno.ns.d.ts:3761]: Maybe leave it as is?

File: runtime/worker.rs [hunk 6]
```diff
@@ -208,6 +209,16 @@ impl MainWorker {
...
 
+    let op_summary_metrics = Rc::new(OpMetricsSummaryTracker::default());
+    deno_core::extension!(deno_op_metrics,
+      options = {
+        op_summary_metrics: Rc<OpMetricsSummaryTracker>
+      },
+      state = |state, options| {
+        state.put(options.op_summary_metrics)
+      }
+    );
```
Comment (bartlomieju) [inline runtime/worker.rs:220]: Is it only for `MainWorker`? What about `WebWorker`?

Also seems strange to define an extension to put a default object in the state. Maybe do it directly after creating ` JsRuntime`?

File: cli/tests/unit/metrics_test.ts (hunk 1 above)
Comment (littledivy) [reply cli/tests/unit/metrics_test.ts:4]: Let's still keep it?

File: runtime/worker.rs (hunk 6 above)
Comment (mmastrac) [reply runtime/worker.rs]: I added code for WebWorker as well -- the metrics there are never fetched (AFAICT) but I suppose it's good for consistency. 

I switched it up to manually add it to the JsRuntime after as suggested but `deno_permissions_worker` is also doing something similar. I'm OK either way.

**Review by bartlomieju — APPROVED** [review]
LGTM once CI passes

**Pull request merged**